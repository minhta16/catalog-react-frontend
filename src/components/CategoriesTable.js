import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Paper, Grid, MenuItem, MenuList, Typography, Divider,
} from '@material-ui/core';

import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchCategories as fetchCategoriesRedux } from 'references/redux/actions/categories';
import { fetchPosts as fetchPostsRedux } from 'references/redux/actions/posts';

class CategoriesTable extends Component {
  state = {};

  componentDidMount() {
    const { fetchCategories, selectedCatId, fetchPosts } = this.props;
    fetchCategories();
    if (selectedCatId) {
      fetchPosts(selectedCatId);
    }
  }

  render() {
    const {
      categories,
      selectedCatId,
      selectedCatItems,
      fetchPosts,
    } = this.props;
    const selectedCatName = categories[selectedCatId] ? categories[selectedCatId].name : '';
    return (
      <Paper>
        <Grid container spacing={2}>
          <Grid item xs={3} className="vertical-divider">
            <Typography variant="h4" className="mainmenu-categories-header">
              Categories
            </Typography>
            <MenuList>
              {Object.keys(categories).map(key => (
                <MenuItem
                  key={key}
                  selected={selectedCatId === key}
                  component={Link}
                  to={`/${key}`}
                  onClick={() => fetchPosts(key)}
                >
                  {categories[key].name}
                </MenuItem>
              ))}
            </MenuList>
          </Grid>
          <Grid item xs={9}>
            <Typography className="left-margin" variant="h4">
              {selectedCatId ? selectedCatName : 'Choose a category'}
            </Typography>
            <Divider variant="middle" />
            <MenuList>
              {selectedCatId ? (
                Object.keys(selectedCatItems).map(key => (
                  <MenuItem key={key}>{selectedCatItems[key].name}</MenuItem>
                ))
              ) : (
                <Typography variant="body1">
                  Please select a category on the left to view its items.
                </Typography>
              )}
            </MenuList>
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

CategoriesTable.propTypes = {
  selectedCatId: PropTypes.string.isRequired,
  categories: PropTypes.shape.isRequired,
  fetchCategories: PropTypes.func.isRequired,
  selectedCatItems: PropTypes.shape.isRequired,
  fetchPosts: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  selectedCatItems: state.posts,
});

const mapDispatchToProps = dispatch => ({
  fetchCategories: () => dispatch(fetchCategoriesRedux()),
  fetchPosts: id => dispatch(fetchPostsRedux(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CategoriesTable);
