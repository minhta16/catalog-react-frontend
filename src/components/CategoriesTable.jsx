import React, { Component } from 'react';
import {
  Paper, Grid, MenuItem, MenuList, Typography, Divider,
} from '@material-ui/core';

class CategoriesTable extends Component {
  state = {
    selectedCategory: { id: -1, name: '' },
    categories: [
      {
        id: 1,
        name: 'Football',
      },
      { id: 2, name: 'Soccer' },
    ],
  };

  componentDidMount() {
    return 0;
  }

  handleItemClick = category => () => {
    this.setState({
      selectedCategory: category,
    });
  };

  render() {
    const { categories, selectedCategory } = this.state;
    return (
      <Paper>
        <Grid container spacing={2}>
          <Grid item xs={3} className="vertical-divider">
            <Typography variant="h4" className="mainmenu-categories-header">
              Categories
            </Typography>
            <MenuList>
              {categories.map(category => (
                <MenuItem
                  key={category.name}
                  selected={selectedCategory.id === category.id}
                  onClick={this.handleItemClick(category)}
                >
                  {category.name}
                </MenuItem>
              ))}
            </MenuList>
          </Grid>
          <Grid item xs={9}>
            <Typography className="left-margin" variant="h4">
              {selectedCategory.name || 'Choose a category'}
            </Typography>
            <Divider variant="middle" />
            <MenuList>
              {categories.map(category => (
                <MenuItem>{category.name}</MenuItem>
              ))}
            </MenuList>
          </Grid>
        </Grid>
      </Paper>
    );
  }
}

export default CategoriesTable;
