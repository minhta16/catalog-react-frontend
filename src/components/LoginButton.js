import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button, Avatar, Menu, MenuItem, Grid, Typography } from '@material-ui/core';

export class LoginButton extends Component {
  state = {
    anchorEl: undefined,
  };

  handleClickAvatar = (e) => {
    this.setState({
      anchorEl: e.target,
    });
  };

  handleCloseMenu = () => {
    this.setState({
      anchorEl: undefined,
    });
  };

  render() {
    const { onClick, variant, color, currentUser } = this.props;
    const { anchorEl } = this.state;
    const menu = (
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={!!anchorEl}
        onClose={this.handleCloseMenu}
        getContentAnchorEl={null}
        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
        transformOrigin={{ vertical: 'top', horizontal: 'center' }}
      >
        <MenuItem onClick={this.handleCloseMenu}>Profile</MenuItem>
        <MenuItem onClick={this.handleCloseMenu}>Logout</MenuItem>
      </Menu>
    );
    return (
      <div>
        {Object.keys(currentUser).length ? (
          <Grid container alignItems="center">
            <Typography variant="body1">{`Welcome, ${currentUser.username}`}</Typography>
            <Avatar
              style={{ backgroundColor: '#fdb600', color: '#fff', margin: '0.5rem' }}
              onClick={this.handleClickAvatar}
            >
              {currentUser.username.charAt(0)}
            </Avatar>
            {menu}
          </Grid>
        ) : (
          <Button id="login-button" onClick={onClick} variant={variant} color={color}>
            Login
          </Button>
        )}
      </div>
    );
  }
}

LoginButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  variant: PropTypes.string,
  color: PropTypes.string,
  currentUser: PropTypes.object.isRequired,
};

LoginButton.defaultProps = {
  variant: 'contained',
  color: 'primary',
};

export default LoginButton;
