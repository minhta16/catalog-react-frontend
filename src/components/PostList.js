import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import PostItem from './PostItem';

export function PostList(props) {
  const { posts } = props;
  return (
    <div style={{ margin: '.5rem' }}>
      <Typography variant="h4" className="center-margin">
        Your Posts
      </Typography>
      {posts.reverse().map((post) => (
        <PostItem key={post.name} post={post} />
      ))}
    </div>
  );
}

PostList.propTypes = {
  posts: PropTypes.array.isRequired,
};

export default PostList;
