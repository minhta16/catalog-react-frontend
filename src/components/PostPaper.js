import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Divider, Paper, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';

export function PostPaper(props) {
  const { header, body, categoryId } = props;
  return (
    <Paper className="left-margin">
      <Button color="primary" component={Link} to={`/${categoryId}`}>
        Back
      </Button>
      <Typography variant="h4">{header}</Typography>
      <Typography variant="subtitle1">By: Minh Ta</Typography>
      <Divider />
      <Typography variant="body1">{body}</Typography>
    </Paper>
  );
}

PostPaper.propTypes = {
  header: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  categoryId: PropTypes.string.isRequired,
};

export default PostPaper;
