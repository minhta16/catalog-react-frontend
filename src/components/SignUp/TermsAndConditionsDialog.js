import React from 'react';
import PropTypes from 'prop-types';

import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Button,
  DialogContentText,
} from '@material-ui/core';

/**
 * This component renders a mock terms and conditions dialog
 */
export function TermsAndConditionsDialog(props) {
  const { open, onClose } = props;
  const terms = `Cras mattis consectetur purus sit amet fermentum.
  Cras justo odio, dapibus ac facilisis in, egestas eget quam.
  Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
  Praesent commodo cursus magna, vel scelerisque nisl consectetur et.`;
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>Terms and Conditions</DialogTitle>
      <DialogContent dividers>
        <DialogContentText>{terms}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>Got it!</Button>
      </DialogActions>
    </Dialog>
  );
}

TermsAndConditionsDialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
};

TermsAndConditionsDialog.defaultProps = {
  onClose: () => {},
  open: true,
};

export default TermsAndConditionsDialog;
