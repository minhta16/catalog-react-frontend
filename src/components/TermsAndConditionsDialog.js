import React from 'react';
import PropTypes from 'prop-types';

import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Button,
  DialogContentText,
} from '@material-ui/core';

export function TermsAndConditionsDialog(props) {
  const { open, onClose } = props;
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>Terms and Conditions</DialogTitle>
      <DialogContent dividers>
        <DialogContentText>
          {[...new Array(50)]
            .map(
              () => `Cras mattis consectetur purus sit amet fermentum.
Cras justo odio, dapibus ac facilisis in, egestas eget quam.
Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
Praesent commodo cursus magna, vel scelerisque nisl consectetur et.`,
            )
            .join('\n')}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>Got it!</Button>
      </DialogActions>
    </Dialog>
  );
}

TermsAndConditionsDialog.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
};

TermsAndConditionsDialog.defaultProps = {
  onClose: () => {},
  open: true,
};

export default TermsAndConditionsDialog;
