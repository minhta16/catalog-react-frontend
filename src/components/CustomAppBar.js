import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Toolbar, IconButton, Typography,
} from '@material-ui/core';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { signIn as signInRedux } from 'references/redux/actions/users';
import LoginButton from './LoginButton';
import LoginDialog from './LoginDialog';

export class CustomAppBar extends Component {
  state = {
    open: false,
  };

  handleLoginClick = () => {
    const { open } = this.state;
    this.setState({
      open: !open,
    });
  };

  render() {
    const { open } = this.state;
    const { color, currentUser, signIn } = this.props;
    return (
      <AppBar color={color}>
        <Toolbar>
          <IconButton edge="start" color="inherit" component={Link} exact="true" to="/">
            <img src="/img/logo.svg" alt="site logo" />
          </IconButton>
          <Typography style={{ flexGrow: '1' }} />
          <LoginButton currentUser={currentUser} onClick={this.handleLoginClick} />
          <LoginDialog open={open} onClose={this.handleLoginClick} onClick={signIn} />
        </Toolbar>
      </AppBar>
    );
  }
}

CustomAppBar.propTypes = {
  color: PropTypes.string.isRequired,
  currentUser: PropTypes.object,
  signIn: PropTypes.func.isRequired,
};

CustomAppBar.defaultProps = {
  currentUser: { username: '', token: '' },
};

const mapStateToProps = state => ({
  currentUser: state.currentUser,
});

const mapDispatchToProps = dispatch => ({
  signIn: (username, password) => {
    dispatch(signInRedux(username, password));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CustomAppBar);
