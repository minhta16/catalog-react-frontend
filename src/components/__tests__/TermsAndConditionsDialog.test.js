/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { TermsAndConditionsDialog } from 'components/TermsAndConditionsDialog';
import { Button } from '@material-ui/core';

describe('components/TermsAndConditionsDialog', () => {
  let wrapper;
  let props;
  // let warning;

  const setup = () => {
    wrapper = shallow(<TermsAndConditionsDialog {...props} />);
  };

  beforeEach(() => {
    props = {
      open: true,
      onClose: jest.fn(),
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClose when click GotIt! button', () => {
    setup();
    const button = wrapper.find(Button);
    button.simulate('click');
    const { onClose } = props;
    expect(onClose).toHaveBeenCalled();
  });
});
