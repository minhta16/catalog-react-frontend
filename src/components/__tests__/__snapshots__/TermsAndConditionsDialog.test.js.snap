// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TermsAndConditionsDialog should render correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Terms and Conditions
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))
    dividers={true}
  >
    <WithStyles(ForwardRef(DialogContentText))>
      Cras mattis consectetur purus sit amet fermentum.
  Cras justo odio, dapibus ac facilisis in, egestas eget quam.
  Morbi leo risus, porta ac consectetur ac, vestibulum at eros.
  Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      onClick={[MockFunction]}
    >
      Got it!
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
